### DISCUSSION TITLE
Should I install @mantine/utils as dependancy to my project?

### DISCUSSION THREAD

**Discussion by Sajarin-M** [mantinedev/mantine#2974] [Q&A] [answered]
I installed `@mantine/utils` as a dependency on my project, for using the `PolymorphicComponentProps` type to create a `CustomButton` extending mantine `Button` with some default props. like this
```
import { FaPlusCircle } from 'react-icons/fa';
import { Button, ButtonProps } from '@mantine/core';
import type { PolymorphicComponentProps } from '@mantine/utils';

type Props = Omit<PolymorphicComponentProps<'button', ButtonProps>, 'leftIcon'>;

export default function AddButton(props: Props) {
  return <Button radius='xl' leftIcon={<FaPlusCircle size='1.1rem' />} {...props} />;
}
```
- Is there anything wrong with best practices in this approach?
- Is `@mantine/utils` something for internal usage only?
- Is there a preferred way to approach this?

**Comment by rtivital** [comment, marked as answer]
You can install and use `@mantine/utils` package, it should work just fine. It does not have documentation though, so you will need to explore source code to find usage examples.
